src/Emogrifier/CssInliner.php

```diff
@@ -660,16 +660,23 @@ public function getMatchingUninlinableSelectors()
      */
     private function determineMatchingUninlinableCssRules(array $cssRules)
     {
-        $this->matchingUninlinableCssRules = \array_filter(
-            $cssRules,
-            function (array $cssRule) {
-                $selector = $cssRule['selector'];
-                if ($cssRule['hasUnmatchablePseudo']) {
-                    $selector = $this->removeUnmatchablePseudoComponents($selector);
-                }
-                return $this->existsMatchForCssSelector($selector);
-            }
-        );
+        $this->matchingUninlinableCssRules = \array_filter($cssRules, [$this, 'existsMatchForSelectorInCssRule']);
+    }
+
+    /**
+     * @param string[] $cssRule
+     *
+     * @return bool
+     *
+     * @throws SyntaxErrorException
+     */
+    private function existsMatchForSelectorInCssRule(array $cssRule)
```

Comment:
The name looks good and is consistent with the naming of the existing method `existsMatchForCssSelector`.

However, I'm wondering whether it should be clarified (probably in the PHPDoc rather than the name - I'm struggling to put this succinctly) that any dynamic pseudo-classes will be assumed to apply, and that if the selector matches a pseudo-element, it will test for a match with its originating element.

Reply:
I have added a PHPDoc description for this.